c_binding/include/libstoragemgmt/libstoragemgmt_plug_interface.h

```diff
@@ -1004,6 +1004,84 @@ typedef int (*lsm_plug_volume_ident_led_set) (lsm_plugin_ptr c,
 typedef int (*lsm_plug_volume_ident_led_clear) (lsm_plugin_ptr c,
                                                 lsm_volume * volume,
                                                 lsm_flag flags);
+/**
+ * Query the RAM cache information of a volume
+ * @param[in]   c               Valid lsm plug-in pointer
+ * @param[in]   volume          Volume to be deleted
+ * @param[out] write_cache_policy
+ *                         The write cache policy. Valid values are:
+ *                          * LSM_VOLUME_WRITE_CACHE_POLICY_WRITE_BACK
+ *                            The storage system will use write back mode if
+ *                            cache hardware found. In write back mode, I/O
+ *                            write data is done asynchronously to the cache and
+ *                            to the backing store(normally disk), the
+ *                            acknowledgement will send back to I/O initiator
+ *                            once cache stored all data.
+ *                          * LSM_VOLUME_WRITE_CACHE_POLICY_AUTO
+ *                            The controller will use write back mode when
+ *                            battery/capacitor is in good health, otherwise,
+ *                            write through mode.
+ *                          * LSM_VOLUME_WRITE_CACHE_POLICY_WRITE_THROUGH
+ *                              The storage system will use write through mode.
+ *                          * LSM_VOLUME_WRITE_CACHE_POLICY_UNKNOWN
+ *                              Plugin failed to detect this setting.
+ * @param[out] write_cache_status
+ *                         The status of write cache. Valid values are:
+ *                          * LSM_VOLUME_WRITE_CACHE_STATUS_WRITE_THROUGH
+ *                          * LSM_VOLUME_WRITE_CACHE_STATUS_WRITE_BACK
+ *                          * LSM_VOLUME_WRITE_CACHE_STATUS_UNKNOWN
+ *
+ * @param[out] read_cache_policy
+ *                          The policy for read cache. Valid values are:
+ *                          * LSM_VOLUME_READ_CACHE_POLICY_ENABLED
+ *                              Read cache is enabled, when reading I/O on
+ *                              previous unchanged written I/O or read I/O in
+ *                              cache will be return I/O initiator immediately
```

Comment:
will be returned to the I/O initiator immediately without checking the backing store